app/controllers/register/establishers/company/director/DirectorNameController.scala

```diff
@@ -40,19 +45,54 @@ class DirectorNameController @Inject()(
                                         requireData: DataRequiredAction,
-                                        formProvider: DirectorNinoFormProvider
+                                        formProvider: PersonNameFormProvider,
+                                        sectionComplete: SectionComplete
                                       )(implicit val ec: ExecutionContext) extends FrontendController with Retrievals with I18nSupport with Enumerable.Implicits {
 
+  private val form = formProvider()
+
+  private def postCall: (Mode, Index, Index, Option[String]) => Call = routes.DirectorNameController.onSubmit _
+
   def onPageLoad(mode: Mode, establisherIndex: Index, directorIndex: Index, srn: Option[String]): Action[AnyContent] =
     (authenticate andThen getData(mode, srn) andThen allowAccess(srn) andThen requireData).async {
       implicit request =>
-        Future.successful(Ok("unimplemented page: " + this.getClass))
+        val preparedForm = request.userAnswers.get[PersonName](DirectorNameId(establisherIndex, directorIndex)) match {
+          case None => form
+          case Some(value) => form.fill(value)
+        }
+        Future.successful(Ok(directorName(
+          appConfig, preparedForm, mode, establisherIndex, directorIndex, existingSchemeName, postCall(mode, establisherIndex, directorIndex, srn), srn)))
     }
 
+  def onSubmit(mode: Mode, establisherIndex: Index, directorIndex: Index, srn: Option[String]): Action[AnyContent] =
+    (authenticate andThen getData(mode, srn) andThen requireData).async {
+      implicit request =>
+        form.bindFromRequest().fold(
+          (formWithErrors: Form[_]) =>
+            Future.successful(BadRequest(directorName(
+              appConfig, formWithErrors, mode, establisherIndex, directorIndex, existingSchemeName, postCall(mode, establisherIndex, directorIndex, srn), srn)))
+          ,
+          value => {
+            val answers = request.userAnswers.set(IsNewDirectorId(establisherIndex, directorIndex))(true).flatMap(
+              _.set(DirectorNameId(establisherIndex, directorIndex))(value)).asOpt.getOrElse(request.userAnswers)
 
-  def onSubmit(mode: Mode, establisherIndex: Index, directorIndex: Index, srn: Option[String]): Action[AnyContent] = (authenticate andThen getData(mode, srn) andThen requireData).async {
-    implicit request =>
-      Future.successful(Ok("unimplemented page: " + this.getClass))
-  }
+            userAnswersService.upsert(mode, srn, answers.json).flatMap {
+              cacheMap =>
+                val userAnswers = UserAnswers(cacheMap)
+                val allDirectors = userAnswers.allDirectorsAfterDelete(establisherIndex)
+                val allDirectorsCompleted = allDirectors.count(_.isCompleted) == allDirectors.size
 
+                if (allDirectorsCompleted) {
+                  Future.successful(Redirect(navigator.nextPage(DirectorNameId(establisherIndex, directorIndex), mode, userAnswers, srn)))
```

Comment:
this condition is no longer needed as we will no longer be using one flag for establisher company completed instead its different spokes